src/index.ts

```diff
@@ -541,19 +542,24 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
       if (id) {
         this.connectionQueueEventEmitter.emit(`connection_${id}`, client);
       } else if (this.options.idleTimeoutMillis > 0) {
-        client.idleTimeoutTimer = setTimeout(() => {
-          this._removeConnection(client);
+        client.idleTimeoutTimer = setTimeout((): void => {
+          // eslint-disable-next-line no-void
+          void this._removeConnection(client);
         }, this.options.idleTimeoutMillis);
 
         this.idleConnections.push(client);
         this.emit('connectionIdle');
       } else {
-        this._removeConnection(client);
+        await this._removeConnection(client);
       }
     };
 
-    client.errorHandler = (err: Error): void => {
-      this._removeConnection(client);
+    // we assign an async promise to a void
+    // because we don't want to enforce a promise signature
+    // on errorHandler
+    // also, EventEmitter.on("error") doesn't support () => Promise<void> (see the on("error"))
+    client.errorHandler = async (err: Error): Promise<void> => {
+      await this._removeConnection(client);
```

Comment:
I do wonder if it is needed that we remove the connection here. If `postgres-pool` uses the `client`, then it'll `release()` in the `finally` block:

https://github.com/postgres-pool/postgres-pool/pull/57/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R467-R469

If the user acquires the connection, they are mandated to call `.release()` themselves!

Reply:
I believe I originally followed how pg-pool handled errors - they terminate and remove the client from the pool. I think that might be ok to leave as the behavior as a catch-all.

Reply:
Will leave it in!